pkg/authorization/registry/role/virtual_registry.go

```diff
@@ -0,0 +1,160 @@
+package role
+
+import (
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	kapierrors "github.com/GoogleCloudPlatform/kubernetes/pkg/api/errors"
+	klabels "github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+
+	authorizationapi "github.com/openshift/origin/pkg/authorization/api"
+	policyregistry "github.com/openshift/origin/pkg/authorization/registry/policy"
+)
+
+type VirtualRegistry struct {
+	policyRegistry policyregistry.Registry
+}
+
+// NewVirtualRegistry creates a new REST for policies.
+func NewVirtualRegistry(policyRegistry policyregistry.Registry) Registry {
+	return &VirtualRegistry{policyRegistry}
+}
+
+// TODO either add selector for fields ot eliminate the option
+func (m *VirtualRegistry) ListRoles(ctx kapi.Context, labels, fields klabels.Selector) (*authorizationapi.RoleList, error) {
+	policyList, err := m.policyRegistry.ListPolicies(ctx, klabels.Everything(), klabels.Everything())
+	if err != nil {
+		return nil, err
+	}
+
+	roleList := &authorizationapi.RoleList{}
+
+	for _, policy := range policyList.Items {
+		for _, role := range policy.Roles {
+			if labels.Matches(klabels.Set(role.Labels)) {
+				roleList.Items = append(roleList.Items, role)
+			}
+		}
+	}
+
+	return roleList, nil
+}
+
+func (m *VirtualRegistry) GetRole(ctx kapi.Context, name string) (*authorizationapi.Role, error) {
+	policy, err := m.policyRegistry.GetPolicy(ctx, authorizationapi.PolicyName)
+	if err != nil && kapierrors.IsNotFound(err) {
+		return nil, kapierrors.NewNotFound("Role", name)
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	role, exists := policy.Roles[name]
```

Comment:
Maybe a hash of the content.  I'm going to claim that as a follow up problem since it will almost certainly ripple through bits of the code that think it should be a number.